src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -0,0 +1,306 @@
+        };
+
+        //! Should be updateRecord...ideally takes this as a prop
+        
+        props.updateFunction(id, recordToUpload);
+    }
+    editingRecord.editing = !editingRecord.editing;
+    let updatedRecordList = recordList.map((record) => {
+        if(record.id === id){
+            return editingRecord;
+        } else {
+            return record;
+        }
+    });
+    console.log(updatedRecordList);
+    setRecordList(updatedRecordList);
+};
+
+const handleEditRecord = (type, id) => (e) => {
+    let editingRecord = recordList.find((record) => {return record.id === id});
+    switch(type) {
+        case "name":
+            editingRecord.name = e.target.value;
+            break;
+        case "description":
+            editingRecord.description = e.target.value;
+            break;
+    }
+    
+
+    //! RecordToUpdateIndex = recordList.indexOf(editingRecord)
+    //! update record... => recordList[indexOf(editingRecord)] = editingRecord
+    let updatedRecordList = recordList.map((record) => {
+        if (record.id == id) {
+            return editingRecord;
+        } else {
+            return record;
+        }
+    });
+    setRecordList(updatedRecordList);
+};
+
+const editRecordRow = (record) => {
+    console.log("edit");
+    return (
+        <Paper square={true} className={"category-row"} >
+            <Grid container alignItems={"center"}>
+                <Grid item xs={3} md={3} >
+                    <TextField
+                        value={record.name}
+                        defaultValue={record.name}
+                        label={"Name"}
+                        onChange={handleEditRecord("name", record.id)}
+                    />
+                </Grid>
+                <Grid item xs={7} md={7}>
+                    <TextField
+                        value={record.description}
+                        defaultValue={record.description}
+                        label={"Description"}
+                        onChange={handleEditRecord("description", record.id)}
+                    />
+                </Grid>
+                <Grid item xs={2} md={2}>
+                    <Button
+                        onClick={editRecord(record.id)}
+                        className={"button"}>
+                            UPDATE
+                        </Button>
+                </Grid>
+            </Grid>
+        </Paper>
+    )
+}
+    const generateGrid = () => {
+        /**
+         * Generates header with fields
+         */
+        
+        const headerElements = [];
+        for (const [index, value] of fieldsWithDefaults.entries()) {
+
+            headerElements.push(
+                <Grid item xs={value["col-width"]} md={value["col-width"]} >
+                    <Typography align={value["align"]} style={{color: 'white', fontWeight: '500'}}>
+                        {value["name"]}
+                    </Typography>
+                </Grid>
+            )
+        }
```

Comment:
For lines 242 - 256, you should separate this to a separate `renderHeader()` function.